src/MongoDB.Analyzer/Core/Builders/ConstantsMapperComposer.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Copyright 2021-present MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License")
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace MongoDB.Analyzer.Core.Builders
+{
+    internal static class ConstantsMapperComposer
+    {
+        public static SpecialType GetSpecialType(object value) =>
+            value switch
+            {
+                sbyte @sbyte => SpecialType.System_SByte,
+                byte @byte => SpecialType.System_Byte,
+                short @short => SpecialType.System_Int16,
+                ushort @ushort => SpecialType.System_UInt16,
+                int @int => SpecialType.System_Int32,
+                uint @uint => SpecialType.System_UInt32,
+                long @long => SpecialType.System_Int64,
+                ulong @ulong => SpecialType.System_UInt64,
+                double @double => SpecialType.System_Double,
+                string @string => SpecialType.System_String,
+                _ => throw new NotSupportedException($"Not supported type {value?.GetType()}")
+            };
+
+        public static void RegisterLiterals(ConstantsMapper newMapper, List<ConstantsMapper> constantsMappers)
```

Comment:
not necessarily "newBuilders" expression, just `node`.
`IEnumerable<ContstantsMapper>`, we don't really use List functionality here.